File: demo_app/www/myblogs/blog_page.js
document.addEventListener("DOMContentLoaded", function () {
  const container = document.getElementById("blogContainer");
  const prevBtn = document.getElementById("prevBtn");
  const nextBtn = document.getElementById("nextBtn");

  // Extract route from URL
  const routeFull = window.location.pathname.split('/');
  const blogRoute = routeFull[2]; // /blogs/<route>

  let currentIndex = 0;
  let blogsList = [];

  function loadBlog(route) {
    container.innerHTML = "<p>Loading blog...</p>";

    frappe.call({
      method: "demo_app.www.myblogs.blog.get_blog_by_route",
      args: { route: route },
      callback: function (r) {
        if (r.message) {
          const blog = r.message.blog;
          blogsList = r.message.all_blogs; // for navigation
          currentIndex = blogsList.findIndex(b => b.name === blog.name);

          container.innerHTML = `
            <h2>${blog.title}</h2>
            <p><b>Author:</b> ${blog.author}</p>
            <p>${blog.content}</p>
            <small>Published on: ${blog.published_on}</small>
          `;

          // Disable buttons if at start or end
          prevBtn.disabled = currentIndex <= 0;
          nextBtn.disabled = currentIndex >= blogsList.length - 1;
        } else {
          container.innerHTML = "<p>Blog not found.</p>";
          prevBtn.disabled = true;
          nextBtn.disabled = true;
        }
      }
    });
  }

  prevBtn.addEventListener("click", function () {
    if (currentIndex > 0) {
      const prevRoute = blogsList[currentIndex - 1].route;
      window.location.href = `/blogs/${prevRoute}`;
    }
  });

  nextBtn.addEventListener("click", function () {
    if (currentIndex < blogsList.length - 1) {
      const nextRoute = blogsList[currentIndex + 1].route;
      window.location.href = `/blogs/${nextRoute}`;
    }
  });

  // Initial load
  loadBlog(blogRoute);
});

File: demo_app/www/myblogs/blog_page.py
import frappe

def get_context(context, route):
    blog = frappe.get_all(
        "Blog_Detail",
        filters={"route": route, "status": "Published"},
        fields=["name", "title", "content", "author", "published_on", "route"]
    )

    if not blog:
        frappe.throw("Blog not found")

    context.blog = blog[0]

    # Fetch previous and next blogs based on published_on
    context.prev_blog = frappe.get_all(
        "Blog_Detail",
        filters=[["published_on", "<", context.blog.published_on], ["status", "=", "Published"]],
        order_by="published_on desc",
        fields=["title", "route"],
        limit_page_length=1
    )
    context.next_blog = frappe.get_all(
        "Blog_Detail",
        filters=[["published_on", ">", context.blog.published_on], ["status", "=", "Published"]],
        order_by="published_on asc",
        fields=["title", "route"],
        limit_page_length=1
    )

    return context


